src/main/java/org/broadinstitute/hellbender/engine/spark/BroadcastJoinReadsWithVariants.java

```diff
@@ -21,13 +22,13 @@
 public final class BroadcastJoinReadsWithVariants {
     private BroadcastJoinReadsWithVariants(){}
 
-    public static JavaPairRDD<GATKRead, Iterable<GATKVariant>> join(final JavaRDD<GATKRead> reads, final JavaRDD<GATKVariant> variants ) {
+    public static JavaPairRDD<GATKRead, Iterable<GATKVariant>> join(final JavaRDD<GATKRead> reads, final JavaRDD<GATKVariant> variants, final List<String> variantsPaths ) {
```

Comment:
It feels like we should have two different methods here, one that takes `JavaRDD<GATKVariant>` and one that takes `List<String>`, instead of having one method that takes one or more of them.     

Also, this method needs javadoc (not your fault that its missing, but would you mind adding it?), particularly if it can take two mutually exclusive inputs.